docs/developer/operator_overview.md

```diff
@@ -0,0 +1,673 @@
+*   `tolerations`: A list of tolerations to allow pods to be scheduled on nodes with matching taints.
+*   `affinity`: A standard Kubernetes `Affinity` object to define complex node and pod affinity/anti-affinity rules.
+
+```yaml
+spec:
+  deploymentPlan:
+    nodeSelector:
+      disktype: ssd
+    tolerations:
+    - key: "app"
+      operator: "Equal"
+      value: "artemis"
+      effect: "NoSchedule"
+    affinity:
+      podAntiAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+        - labelSelector:
+            matchExpressions:
+            - key: "app"
+              operator: In
+              values:
+              - artemis
+          topologyKey: "kubernetes.io/hostname"
+```
+
+### Applying Custom Patches with Resource Templates
+
+For advanced customizations that are not directly exposed in the CRD, you can use `resourceTemplates` to apply a strategic merge patch to the resources the operator creates (like the `StatefulSet`).
+
+This example uses a patch to add a custom annotation to the broker's `StatefulSet`.
+
+```yaml
+spec:
+  resourceTemplates:
+    - selector:
+        kind: StatefulSet
+      patch:
+        metadata:
+          annotations:
+            "my-custom-annotation": "my-custom-value"
+```
+
+## 8. Metrics and Monitoring with Prometheus
+
+The operator supports exposing broker metrics in a Prometheus-compatible format. This is controlled by the `enableMetricsPlugin` flag in the custom resource.
+
+### Enabling Metrics
+
+To enable metrics, set `spec.deploymentPlan.enableMetricsPlugin: true` in your `ActiveMQArtemis` CR:
+
+```yaml
+apiVersion: broker.amq.io/v1beta1
+kind: ActiveMQArtemis
+metadata:
+  name: ex-aao
+spec:
+  deploymentPlan:
+    enableMetricsPlugin: true
+    # ... other deployment plan fields ...
+```
+
+### Deployment Modes
+
+The way metrics are exposed depends on whether the broker is deployed in `restricted` mode.
+
+#### Legacy Mode (default)
+
+When `restricted: false` (or is omitted), enabling the metrics plugin adds the official [Artemis Prometheus Metrics Plugin](https://activemq.apache.org/components/artemis/documentation/latest/metrics.html) to the broker's classpath.
+
+*   **Endpoint**: Metrics are exposed on the broker's web console port (default `8161`) at the `/metrics` path.
+*   **Security**: The endpoint is secured by the same credentials as the web console.
+
+#### Restricted (Locked-Down) Mode
+
+When `restricted: true`, the operator injects the [Prometheus JMX Exporter](https://github.com/prometheus/jmx_exporter) as a Java agent into the broker pod.
+
+*   **Endpoint**: Metrics are exposed on a dedicated port (default `8888`) at the `/` path.
+*   **Security**: The endpoint is secured with mTLS, and clients must present a valid certificate. By default, only clients with a certificate CN of `activemq-artemis-operator` are authorized, but this can be customized with `brokerProperties`.
```

Comment:
cannot be modified at the moment. it would need an update to the login.config key in the properties secret, to which there is currently no access.